pandas/tests/extension/decimal/array.py

```diff
@@ -130,8 +131,13 @@ def copy(self):
         return type(self)(self._data.copy())
 
     def astype(self, dtype, copy=True):
+        from pandas.core.arrays.string_ import StringDtype
+
+        dtype = pandas_dtype(dtype)
```

Comment:
I think this here should be the array/base.py impl.

Reply:
can you do this

Reply:
> I think this here should be the array/base.py impl.

The base class version already does this?

Reply:
I've replaced that with a ``super``-call.

Reply:
see my comments above

Reply:
The decimal astype implementation needs to handle the "context", so can't be fully moved to the base class (@topper-123 already moved what can be moved and calls `super()` in the end.